inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_sequences_to_sequences_ie.cpp

```diff
@@ -66,7 +109,13 @@ ngraph::pass::ConvertLSTMSequenceMatcher::ConvertLSTMSequenceMatcher() {
         unsqueeze_1->set_friendly_name(lstm_sequence->get_friendly_name()+".0");
         unsqueeze_2->set_friendly_name(lstm_sequence->get_friendly_name()+".1");
         unsqueeze_3->set_friendly_name(lstm_sequence->get_friendly_name()+".2");
-        ngraph::replace_node(lstm_sequence, {unsqueeze_1->output(0), unsqueeze_2->output(0), unsqueeze_3->output(0)});
+        if (seq_axis == 1) {
+            ngraph::replace_node(lstm_sequence, {unsqueeze_1->output(0), unsqueeze_2->output(0), unsqueeze_3->output(0)});
+        } else {
+            auto transpose_after = lstm_sequence->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
```

Comment:
We don't change operations that are above matched nodes cause it leads to undefined behavior.
I believe we can make a pattern that will match LSTMSequence->AnyOp and then you will work with AnyOp directly.

Reply:
ticket 43346